### DISCUSSION TITLE
Understanding multiparty encryption

### DISCUSSION THREAD

**Discussion by uo257724** [tuneinsight/lattigo#286] [Q&A] [answered]
I'm trying to understand how multiparty encryption works in detail, I understand that the pary encrypt with a collective public key and for decryption, anyone in the party can use his secret key for it. is this correct? 
Secondly, when the decryption comes from an external party, the key used for decryption is generated by the party for this purpose. Is this also correct?
The example I've been reading is [this](https://github.com/tuneinsight/lattigo/blob/master/examples/dbfv/pir/main.go)

Thank you all in advance.

**Comment by ChristianMct** [comment, marked as answer]
> I'm trying to understand how multiparty encryption works in detail

It is probably easier to start from theoretical material to understand how multiparty HE works. Our[ PETS'21 paper ](https://eprint.iacr.org/2020/304.pdf) on the topic is a good resource for that. See also [the video of its presentation](https://youtu.be/x0MXA8zL-Q8) for a more condensed version.

> I understand that the pary encrypt with a collective public key and for decryption, anyone in the party can use his secret key for it. is this correct?
No. When encrypting with the collective encryption-key, *ALL* parties in the set need to collaborate in order to decrypt (when using the basic threshold setting as in the mentioned PIR example).

> Secondly, when the decryption comes from an external party, the key used for decryption is generated by the party for this purpose. Is this also correct?

Assuming that you are referring to external receiver parties (as defined in the sense of the paper), yes. The receiver party generates a public-private encryption key pair and sends the public one to the parties. The parties then re-encrypt the messages from the collective key to the receiver key.

Hope this helps.
